coderd/workspaceagents.go

```diff
@@ -152,6 +153,50 @@ func (api *API) workspaceAgentMetadata(rw http.ResponseWriter, r *http.Request)
 	})
 }
 
+func (api *API) postWorkspaceAgentVersion(rw http.ResponseWriter, r *http.Request) {
+	workspaceAgent := httpmw.WorkspaceAgent(r)
+	apiAgent, err := convertWorkspaceAgent(workspaceAgent, nil, api.AgentInactiveDisconnectTimeout)
+	if err != nil {
+		httpapi.Write(rw, http.StatusInternalServerError, codersdk.Response{
+			Message: "Internal error reading workspace agent.",
+			Detail:  err.Error(),
+		})
+		return
+	}
+
+	var req codersdk.PostWorkspaceAgentVersionRequest
+	if err := json.NewDecoder(r.Body).Decode(&req); err != nil {
+		httpapi.Write(rw, http.StatusInternalServerError, codersdk.Response{
+			Message: "Invalid request to set workspace agent version.",
+			Detail:  err.Error(),
+		})
+		return
+	}
```

Comment:
I think this is the new fancy way to do this.

```suggestion
	if !httpapi.Read(rw, r, &req) {
		return
	}
```